### PR TITLE
FEM-2673 Add more analytics events properties

### PR DESCRIPTION
**Pull request by giladna**
Make OKHTTP the default http requests library

Added 2 new events 
        CONNECTION_ACQUIRED,
        OUTPUT_BUFFER_COUNT_UPDATE

### CODE REVIEW

File: playkit/src/main/java/com/kaltura/playkit/Utils.java [hunk 1]
```diff
@@ -203,4 +213,48 @@ public static boolean equals(Object a, Object b) {
     public static String format(String format, Object... args) {
         return String.format(Locale.ROOT, format, args);
     }
+
+    public static String getNetworkClass(Context context) {
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/Utils.java:220]: Some of the constants are only available in latest SDKs, won't it cause a crash?

File: playkit/src/main/java/com/kaltura/playkit/player/ExoAnalyticsAggregator.java [hunk 2]
```diff
@@ -52,5 +84,55 @@ public void onLoadError(EventTime eventTime, MediaSourceEventListener.LoadEventI
...
+//    public void onLoadStarted(EventTime eventTime, MediaSourceEventListener.LoadEventInfo loadEventInfo, MediaSourceEventListener.MediaLoadData mediaLoadData) {
+//        String loadedURL = loadEventInfo.uri.toString();
+//        log.v("onLoadStarted = " + eventTime.realtimeMs + " url = " + loadedURL);
+//        if (domainCallStartRelTimeMap != null && mediaLoadData.trackType != C.TRACK_TYPE_VIDEO && domainCallStartRelTimeMap != null && mediaLoadData.trackType != C.TRACK_TYPE_AUDIO && mediaLoadData.trackType != C.TRACK_TYPE_DEFAULT) {
+//            if (domainCallStartRelTimeMap.containsKey(loadedURL)) {
+//                domainCallStartRelTimeMap.remove(loadedURL);
+//                //log.v("onLoadStarted ignore url = " + loadedURL);
+//            }
+//        }
+//    }
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/player/ExoAnalyticsAggregator.java]: Remove....

File: playkit/src/main/java/com/kaltura/playkit/player/ExoAnalyticsAggregator.java [hunk 3]
```diff
@@ -1,20 +1,39 @@
...
+import okhttp3.Connection;
+import okhttp3.EventListener;
+
+import static com.kaltura.playkit.utils.Consts.HTTP_METHOD_GET;
 
-class ExoAnalyticsAggregator implements AnalyticsListener {
+class ExoAnalyticsAggregator extends EventListener implements AnalyticsListener {
 
+    private static final PKLog log = PKLog.get("ExoAnalyticsAggregator");
+    private final Map<String, Long>  domainCallStartRelTimeMap = new ConcurrentHashMap<>();
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/player/ExoAnalyticsAggregator.java]: Why "domain"? Isn't it a map of URLs?

File: playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java [hunk 4]
```diff
@@ -406,9 +406,13 @@ private MediaSource buildExternalSubtitleSource(int subtitleId, PKExternalSubtit
...
-            final okhttp3.EventListener.Factory okListenerFactory = profiler.getOkListenerFactory();
-            if (okListenerFactory != null) {
-                builder.eventListenerFactory(okListenerFactory);
+            if (profiler != Profiler.NOOP) {
+                final okhttp3.EventListener.Factory okListenerFactory = profiler.getOkListenerFactory();
+                if (okListenerFactory != null) {
+                    builder.eventListenerFactory(okListenerFactory);
+                }
+            } else {
+                builder.eventListener(analyticsAggregator);
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java]: Please check if there can be more than one listener -- then we will be able to have both profiler and analytics aggregator.

File: playkit/src/main/java/com/kaltura/playkit/PlayerEvent.java [hunk 5]
```diff
@@ -253,13 +257,63 @@ public String toString() {
         }
     }
 
+    public static class ConnectionAcquired extends PlayerEvent {
+
+        public final long connectDurationMs;
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/PlayerEvent.java]: Let's try to keep 3 values here, if it's not too complicated: DNS duration, connect duration, TLS duration.

**Review by noamtamim — CHANGES_REQUESTED** [review]

File: playkit/src/main/java/com/kaltura/playkit/Utils.java (hunk 1 above)
Comment (giladna) [reply playkit/src/main/java/com/kaltura/playkit/Utils.java:220]: tested this code with 4.1.2 device do you think it is enough?

Comment (giladna) [reply playkit/src/main/java/com/kaltura/playkit/Utils.java:220]: tested and seems fine you cannot print it but it is fine in run time

File: playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java (hunk 4 above)
Comment (giladna) [reply playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java]: move the  builder.eventListener(analyticsAggregator); to be first without else

File: playkit/src/main/java/com/kaltura/playkit/PlayerEvent.java (hunk 5 above)
Comment (giladna) [reply playkit/src/main/java/com/kaltura/playkit/PlayerEvent.java]: combine all to one object and updated the whole chain

**Review by OrenMe — APPROVED** [review]

**Review by noamtamim — APPROVED** [review]

**Pull request merged**